pkg/infra/initiator.go

```diff
@@ -0,0 +1,35 @@
+package infra
+
+import (
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/time/rate"
+)
+
+type Initiator struct {
+}
+
+func (i *Initiator) Start(num int, rateC float64, config Config, crypto *Crypto, raw chan *Elements, errorCh chan error, logger *log.Logger) {
+	limiter := rate.NewLimiter(rate.Inf, size)
+	if rateC > 0 {
+		logger.Infof("Will use rate %f as send rate\n", rateC)
+		limit := rate.Limit(rateC)
+		limiter = rate.NewLimiter(limit, size)
+	}
+	for i := 0; i < num; i++ {
+		if limiter.Allow() {
+			prop, err := CreateProposal(
+				crypto,
+				config.Channel,
+				config.Chaincode,
+				config.Version,
+				config.Args...,
+			)
+			if err != nil {
+				errorCh <- errors.Wrapf(err, "error creating proposal")
+				return
+			}
```

Comment:
tiny improvement - proposal signing could be done in a separate goroutine, right? (or at least not before waiting for tokens)

Reply:
Cool